sof_remove.sh: Remove all ALSA modules

ujfalusi opened the pull request:
Remove the remaining ALSA modules to have completely fresh start when
inserting the modules.
This way we can also test module parameters for the core.

Signed-off-by: Peter Ujfalusi <peter.ujfalusi@linux.intel.com>

marc-hb commented:
https://sof-ci.01.org/softestpr/PR812/build931/devicetest failed with `rmmod: ERROR: Module snd_timer is in use by: snd_seq`

marc-hb commented:
>  rmmod: ERROR: Module snd_timer is in use by: snd_seq

Ping @ujfalusi ?

Is this something you'd like?

ujfalusi commented:
> > rmmod: ERROR: Module snd_timer is in use by: snd_seq
> 
> Ping @ujfalusi ?
> 
> Is this still something you'd like?

@marc-hb, yes, certainly, let me re-spin it.
Thanks for reminding me ;)

ujfalusi commented:
Changes since v1:
- added more modules to be removed (snd_seq_midi, snd_seq_midi_event, snd_rawmidi, snd_seq, snd_seq_device, snd_usbmidi_lib)

If we still see failure then we might want to drop this as we can not list all modules we might encounter...

marc-hb commented:
> If we still see failure then we might want to drop this as we can not list all modules we might encounter...

In that case can you please isolate all these into a new section with some "ALSA modules" title + this comment?

ujfalusi commented:
> https://sof-ci.01.org/softestpr/PR812/build959/devicetest/ is all green now.
> 
> > If we still see failure then we might want to drop this as we can not list all modules we might encounter...
> 
> In that case can you please isolate all these into a new section with some "ALSA modules" title + this comment?

It is passing now in CI so we are covering nicely the common modules, but if you have some new device spanning out new module then it will stop working as we are aiming to take down the whole audio subsystem now.

greg-intel commented:
Is there a new device checklist?  We'd have to add an entry to this file to prevent failure if the new device brings a unique module with it? Or am I not understanding?

marc-hb reviewed (changes requested):
> Is there a new device checklist? We'd have to add an entry to this file to prevent failure if the new device brings a unique module with it? Or am I not understanding?

Yes I think we just grow this list as needed. For me what's missing now is a very brief summary of this discussion in the code itself.

marc-hb commented:
Unlikely conflict with unrelated 2ff9c0bbf13ff8d, can you please rebase?

Also, please summarize the discussion above in the code.

marc-hb commented:
@ujfalusi can you please:
- rebase and fix the conflict
- Add some very short comments summarizing the discussion above

ujfalusi commented:
Changes since v2:
- rebased on main
- 2.5 years passed ;)

marc-hb reviewed (commented):
LGTM but someone familiar with modules must approve. @plbossart , @bardliao maybe?

The LNL overruns in https://sof-ci.01.org/softestpr/PR812/build656/devicetest/index.html are interesting but unrelated. Everything else is either green or not available.

EDIT: overruns filed in new https://github.com/thesofproject/linux/issues/ 5109

plbossart reviewed (approved):
LGTM
